packages/py-api/py_api/controllers/hackathon_teams_controller.py

```diff
@@ -0,0 +1,211 @@
+import json
+import random
+import string
+from typing import Any, Dict, List, Optional
+
+from bson.json_util import dumps
+from bson.objectid import ObjectId
+from fastapi.responses import JSONResponse
+from py_api.database.initialize import t_col
+from py_api.models.hackathon_teams_models import (
+    HackathonTeam,
+    MoveTeamMembers,
+    TeamType,
+    UpdateTeam,
+)
+from py_api.utilities.parsers import filter_none_values
+
+
+class TeamsController:
+    @classmethod
+    def create_team(cls, team_name: str, user_id: str = "", members_list: Optional[List[str]] = None) -> (
+            Dict[str, Any] | None
+    ):
+
+        team_type = TeamType.NORMAL
+
+        if not team_name:
+            # If no team_name is provided during registration, the participant is registering individually.
+            # We create a team of type RANDOM which should be filled with such participants
+
+            team_name = ''.join(
+                random.choice(string.ascii_letters)
+                for _ in range(8)
+            )
+            team_type = TeamType.RANDOM
+
+        team = cls.check_if_team_exists(team_name)
+        team_members = []
+
+        if user_id:
+            team_members = [user_id]
+
+        if members_list:
+            team_members = members_list
+            team_type = TeamType.RANDOM
+
+        if not team:
+            new_team = HackathonTeam(
+                team_name=team_name,
+                team_type=team_type, team_members=team_members,
+            )
+            t_col.insert_one(new_team.model_dump())
+
+            return None
+
+        return team
+
+    @classmethod
+    def add_team_participant(cls, team_name: str, user_id: str) -> List[str] | None:
+        team = cls.check_if_team_exists(team_name)
```

Comment:
1. rename this method to `fetch_team` or something along those lines. I expect this method to return a boolean value, not an object of type Team. You would expect that if a team is not found, nothing is returned. Just makes sense, but not when it's named like that.

2. Let team be `team = HackathonTeam(team_members=team["team_members"], ...)`
This will allow you to easily access the object members and have proper type safety.

Reply:
@IvanObreshkov it seems I've made a mistake here. A method called fetch_team should either return the object or none, not a Boolean value.